crates/codebase_files/src/lib.rs

```diff
@@ -1,41 +1,53 @@
-use std::path::PathBuf;
-use std::process::{Command, Output};
+use ignore::{WalkBuilder, WalkState};
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
 
+#[derive(Debug, PartialEq)]
 pub struct CodebaseFiles {
     pub paths: Vec<PathBuf>,
 }
 
 impl CodebaseFiles {
     pub fn all() -> CodebaseFiles {
-        let output = Command::new("git").arg("ls-files").output();
-        let mut paths = Self::process_ls_files(output);
-        paths.extend(Self::process_ls_files(
-            Command::new("git")
-                .arg("ls-files")
-                .arg("--others")
-                .arg("--exclude-standard")
-                .output(),
-        ));
+        let mut builder = WalkBuilder::new("./");
+        builder.hidden(false);
+        builder.filter_entry(|entry| entry.path() != Path::new("./.git"));
 
+        let results = Arc::new(Mutex::new(vec![]));
+
+        builder.build_parallel().run(|| {
```

Comment:
This should be faster than `build`, but it also means we need to use the arc mutex as the accumulator for paths.